### PR TITLE
Added API to update facilitators screens

### PR DESCRIPTION
**Pull request by parthgandhitekdi**

### CODE REVIEW

File: src/src/facilitator/facilitator.service.ts [hunk 1]
```diff
@@ -56,8 +55,366 @@ export class FacilitatorService {
...
+      'qualification_reference_document_id'
+    ];
+    let keyExist = qualificationsArr.filter((e) => Object.keys(body).includes(e));
+    const qualificationDetails = facilitatorUser.qualifications[0];
+    if (keyExist.length) {
+      const tableName = 'qualifications';
+      await this.hasuraService.q(
+        tableName,
+        {
+          qualification_master_id: body.qualification_master_id,
```
Comment (sagarkoshti1990) [inline src/src/facilitator/facilitator.service.ts:287]: spread body element

File: src/src/facilitator/facilitator.service.ts [hunk 2]
```diff
@@ -383,4 +740,203 @@ export class FacilitatorService {
...
+          profile_url
+          state
+          district
+          block
+          village
+          grampanchayat
+          alternative_mobile_number
+          profile_photo_1
+          profile_photo_2
+          profile_photo_3
```
Comment (sagarkoshti1990) [inline src/src/facilitator/facilitator.service.ts:775]: add document relation ship data

File: src/src/facilitator/facilitator.service.ts [hunk 3]
```diff
@@ -56,8 +55,366 @@ export class FacilitatorService {
...
+      'device_type',
+    ]
+    keyExist = coreFacilitatorsArr.filter((e) => Object.keys(body).includes(e));
+    if (keyExist.length) {
+      const tableName = 'core_faciltators';
+      await this.hasuraService.q(
+        tableName,
+        {
+          ...body,
+          id: facilitatorUser?.core_faciltator?.id ?? null,
```
[the remark this answers is not in the document]
Comment (parthgandhitekdi) [reply src/src/facilitator/facilitator.service.ts:101]: Not getting multiple entries

File: src/src/facilitator/facilitator.service.ts [hunk 4]
```diff
@@ -56,8 +55,366 @@ export class FacilitatorService {
...
+    let keyExist = userArr.filter((e) => Object.keys(body).includes(e));
+    if (keyExist.length) {
+      const tableName = 'users';
+      body.id = id;
+      await this.hasuraService.q(tableName, body, userArr, true);
+    }
+
+    // Update Extended Users table data
+    const extendedUserArr = [
+      'user_id',
```
[the remark this answers is not in the document]
Comment (parthgandhitekdi) [reply src/src/facilitator/facilitator.service.ts]: Done

File: src/src/facilitator/facilitator.service.ts [hunk 5]
```diff
@@ -56,8 +55,366 @@ export class FacilitatorService {
...
+          id: facilitatorUser?.extended_users?.[0]?.id ?? null,
+          user_id: id,
+        },
+        extendedUserArr,
+        true
+      );
+    }
+  }
+
+  async updateWorkExperienceDetails(id: number, body: any, facilitatorUser: any) {
```
[the remark this answers is not in the document]
Comment (parthgandhitekdi) [reply src/src/facilitator/facilitator.service.ts:161]: Done

File: src/src/facilitator/facilitator.service.ts [hunk 6]
```diff
@@ -56,8 +55,366 @@ export class FacilitatorService {
...
+      'experience_in_years',
+      'related_to_teaching',
+      'user_id',
+      'type'
+    ];
+    let keyExist = experienceArr.filter((e) => Object.keys(body).includes(e));
+    let experienceInfo;
+    if (keyExist.length) {
+      const tableName = 'experience';
+      body.id = id;
```
[the remark this answers is not in the document]
Comment (parthgandhitekdi) [reply src/src/facilitator/facilitator.service.ts]: Done

File: src/src/facilitator/facilitator.service.ts (hunk 1 above)
Comment (parthgandhitekdi) [reply src/src/facilitator/facilitator.service.ts]: Done

File: src/src/facilitator/facilitator.service.ts (hunk 2 above)
Comment (parthgandhitekdi) [reply src/src/facilitator/facilitator.service.ts:767]: Done

**Pull request merged**